### ISSUE TITLE
Angular 1.4.0 incorrectly evaluates some expressions

### ISSUE THREAD

**Issue by szimek** [angular/angular.js#12020]
Here's an example to reproduce this issue: http://plnkr.co/edit/XJlfdY1B0gEuepxpgiku?p=preview

The controller has defined these properties:
```javascript
this.onThis = true;
$scope.onScope = {};
```

and in the template the following expression is evaluated to true, even though it shouldn't:
```html
<div ng-controller="MyController as myCtrl">
  <p ng-if="myCtrl.onThis && onScope.some.property">1. This is visible, but should not be, {{name}}!</p>
</div>
```
Switching the order of conditions (i.e. `onScope.some.property && myCtrl.onThis`) surprisingly fixes this issue. It worked correctly in Angular 1.3.x.

**Comment by gkalpak** [comment]
This has been [fixed on master][1]. (I think it's #11989.)

[1]: http://plnkr.co/edit/8KGo7WQxMS06Ys3sM6an?p=preview

**Comment by szimek** [comment]
@gkalpak Thanks! Will `master` branch become 1.4.1 release or is it safer to use `g3_v1_4` branch instead of the current official 1.4.0 release? This issue is really annoying, because our tests can't find all places where it occurs in our templates, so I'd like to use some version that has it fixed while waiting for 1.4.1 release.

**Comment by gkalpak** [comment]
@szimek, `master` will eventually become `v1.4.1` (pretty soon I suspect).
(I'm not sure what `g3_v1_4` is for, but I think it's internal to Google.)

In the mean time, you could use the latest build (which is **not** and officially supported build yet, but has all the latest fixes/improvements). You can grab it from https://github.com/angular/bower-angular or https://code.angularjs.org/snapshot (on your own risk, of course :smiley:).
